packages/core/src/version-utils.ts

```diff
@@ -0,0 +1,106 @@
+import { normalizeErrorString } from './outcomes';
+
+export const GITHUB_RELEASES_URL = 'https://api.github.com/repos/medplum/medplum/releases';
+
+export type ReleaseManifest = { tag_name: string; assets: { name: string; browser_download_url: string }[] };
+
+const releaseManifests = new Map<string, ReleaseManifest>();
+
+/**
+ * Clears the locally-cached `ReleaseManifest`s for all versions.
+ */
+export function clearReleaseCache(): void {
+  releaseManifests.clear();
+}
+
+/**
+ * Asserts that a given candidate is a `ReleaseManifest`.
+ * @param candidate - An object assumed to be a `ReleaseManifest`.
+ */
+export function assertReleaseManifest(candidate: unknown): asserts candidate is ReleaseManifest {
```

Comment:
Do we need all of this validation code?  It seems like we should be able to get the values we need, and either they are present or not?

Reply:
What's wrong with asserting that a particular object has all the appropriate fields? It makes it so that we don't have to check in multiple places. Also some of our releases coming from Github didn't have some of these fields but had others, this was a simple abstraction that ensures we have a valid release manifest past a certain point, and then you get type-time info if something matches this shape.